modules/ml/src/main/java/org/apache/ignite/ml/composition/boosting/GDBLearningStrategy.java

```diff
@@ -157,10 +161,11 @@
     /**
      * Sets learning environment.
      *
-     * @param environment Learning Environment.
+     * @param envBuilder Learning Environment.
      */
-    public GDBLearningStrategy withEnvironment(LearningEnvironment environment) {
-        this.environment = environment;
+    public GDBLearningStrategy withEnvironmentBuilder(LearningEnvironmentBuilder envBuilder) {
```

Comment:
@artemmalykh, could you please have a look at this problem?

Reply:
Perhaps.

Reply:
For now added check, as you asked because otherwise to be consistent, we should probably move all other necessary args into constructor, which I do not want to do in this PR.